File: backend/ChessPlatform.Api/Models/Game.cs
namespace ChessPlatform.Api.Models;

public class Game
{
    public Guid Id { get; set; }
    public Guid WhitePlayerId { get; set; }
    public Guid BlackPlayerId { get; set; }
    public string TimeControl { get; set; } = string.Empty;
    public string Result { get; set; } = string.Empty;
    public string Termination { get; set; } = string.Empty;
    public string Pgn { get; set; } = string.Empty;
    public string FinalFen { get; set; } = string.Empty;
    public int WhiteRatingBefore { get; set; }
    public int BlackRatingBefore { get; set; }
    public int WhiteRatingAfter { get; set; }
    public int BlackRatingAfter { get; set; }
    public DateTime StartedAt { get; set; }
    public DateTime EndedAt { get; set; }
    public User WhitePlayer { get; set; } = null!;
    public User BlackPlayer { get; set; } = null!;
}

File: frontend/src/types/game.ts
export interface GameHistoryEntry {
  id: string;
  opponentName: string;
  opponentAvatar: string | null;
  playerColor: 'white' | 'black';
  playerResult: 'win' | 'loss' | 'draw';
  termination: string;
  timeControl: string;
  ratingBefore: number;
  ratingAfter: number;
  endedAt: string;
}

export interface GameHistoryResponse {
  data: {
    games: GameHistoryEntry[];
    totalCount: number;
    page: number;
    pageSize: number;
  };
}

export interface GameDetail {
  id: string;
  whiteName: string;
  whiteAvatar: string | null;
  blackName: string;
  blackAvatar: string | null;
  timeControl: string;
  result: string;
  termination: string;
  pgn: string;
  finalFen: string;
  whiteRatingBefore: number;
  whiteRatingAfter: number;
  blackRatingBefore: number;
  blackRatingAfter: number;
  startedAt: string;
  endedAt: string;
}
